sklearn/utils/validation.py

```diff
@@ -416,6 +416,11 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         Raise DataConversionWarning if the dtype of the input data structure
         does not match the requested dtype, causing a memory copy.
 
+        .. deprecated:: 0.20
+           warn_on_dtype' is deprecated from 'check_array' in
+           version 0.20 and will be removed in 0.22. Use
+           warn_on_dtype = False' instead.
```

Comment:
should be versions 0.21 - 0.23

No need to precise "from check_array"

Also the tip should indicate `set warn_on_dtype=None to remove the warning`